WordPress/src/main/java/org/wordpress/android/ui/posts/PageAndPostListPreviewStateHelper.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.posts
+
+import androidx.fragment.app.FragmentActivity
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.ui.posts.RemotePreviewLogicHelper.RemotePreviewType
+import org.wordpress.android.ui.uploads.UploadService
+import org.wordpress.android.util.AppLog
+
+fun managePreviewStateTransitions(
+    newState: PostListRemotePreviewState,
+    prevState: PostListRemotePreviewState?,
+    postInfo: PostInfoType,
+    handleRemotePreview: (localPostId: Int, RemotePreviewType) -> Unit
+) = when (newState) {
+    PostListRemotePreviewState.PREVIEWING -> {
+        prevState?.let {
+            if (it == PostListRemotePreviewState.UPLOADING_FOR_PREVIEW ||
+                    it == PostListRemotePreviewState.REMOTE_AUTO_SAVING_FOR_PREVIEW) {
+                (postInfo as? PostInfoType.PostInfo)?.let { info ->
```

Comment:
Do we want to simply ignore the invocation when the client provides a wrong PostInfo object? I think we should consider either 
- assume we have the correct object
`handleRemotePreview((postInfo as PostInfoType.PostInfo).post.id, getRemotePreviewType(prevState))`
- throw an `InvalidArgumentException` when the cast fails
- log an error message

I'm a fan of fail fast principle so I'd personally go with either 1 or 2. 

Wdyt?

Reply:
Updated in fa5c0dee700a549a22f557e3abdef5efa8e4b9a6